setup.py

```diff
@@ -77,15 +95,76 @@ def finalize_options(self):
         ]
         super().finalize_options()
 
+    def add_optional_flags(self):
+        env = os.environ.copy()
+        if sys.platform == 'win32':
+            return env
+
+        cppflags = []
+        if 'CPPFLAGS' in os.environ:
+            cppflags.append(os.environ['CPPFLAGS'])
+        cxxflags = []
+        if 'CXXFLAGS' in os.environ:
+            cxxflags.append(os.environ['CXXFLAGS'])
```

Comment:
AFAICT distutils does not read this; is it passed to anyone?

Reply:
It's returned in the new environment, so it can be applied to external things, like FreeType, or qhull.

Reply:
I guess I was confused, but perhaps you can document this a bit better?  even just naming the thing e.g. `get_env_for_custom_build()` would be enough, I guess

Reply:
This is all in `BuildExtraLibraries`, so that's a little bit redundant.

Reply:
I added a docstring, but also note that `extra_compile_args` _is_ used for internal extensions. The environment variables are for the vendored code.